### ISSUE TITLE
destroy method bug

### ISSUE THREAD

**Issue by johntom** [balderdashy/sails-mongo#219]
Hi,
I recently found a bug(?) where my bug caused the complete deletion of a collection.
1) I wanted to delete all assocated billing records after deleting the contract. 
2) My bug passed an undefined value (in this example var bcs = model.BCSNumber ) to the following and completely deleted the Billing collection

           var bcs = model.BCSNumber;// if this is undefined then it will delete entire collection below
            Contract.destroy(id, function (err) {
                if (err) {
                    return res.serverError(err);
                }
                Billing.destroy({'BCSNumber': bcs}, function (berr) {
                    })
                Contract.publishDestroy(model.id);
                return res.json(model);
            });
TIA
John

**Comment by sailsbot** [comment]
Thanks for posting, @johntom.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
